packages/sdk/src/core/msGraphAuthProvider.ts

```diff
@@ -6,14 +6,15 @@ import { ErrorWithCode, ErrorCode } from "./errors";
 import { TeamsFxConfiguration } from "../models/teamsfxConfiguration";
 import { internalLogger } from "../util/logger";
 import { validateScopesType } from "../util/utils";
+import { AccessToken, TokenCredential } from "@azure/identity";
 
 const defaultScope = "https://graph.microsoft.com/.default";
 
 /**
  * Microsoft Graph auth provider for Teams Framework
  */
 export class MsGraphAuthProvider implements AuthenticationProvider {
-  private teamsfx: TeamsFxConfiguration;
+  private credentialOrTeamsFx: TokenCredential | TeamsFxConfiguration;
```

Comment:
What if having two variables with different priority when used?
```
private teamsfx?: TeamsFxConfiguration;
private credential?: TokenCredential;
...
token = credential ? credential.xxx : teamsfx.xxx;
```

I'm not sure which is the best practice in TS, but just feel the union one may cause more maintenance effort.

Reply:
Yes, you're right, union one may cause more maintenance effort, however it only target for non-breaking change for teamsfx 2.0, and, and in the future for the split SDK, TeamsFx class will be deprecated and remove in the future as discussed.

So it is better not to mixed them together to make it easy for future breaking change SDK.

Reply:
Good to know. (BTW, I like the split SDK as well :D )